### PR TITLE
Implemented DeterministicOldestWhenJoiningSpec

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
Fixed https://github.com/akkadotnet/akka.net/issues/2082

Should go after these two: https://github.com/akkadotnet/akka.net/pull/2098 and https://github.com/akkadotnet/akka.net/pull/2099

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
You can reset the git timestamp using this technique: http://stackoverflow.com/a/31540373/377476 - that way CI can pick up the commit and run.

**Comment by Aaronontheweb** [comment]
Take a look at the failed cluster domain event specs in the unit tests - looks like there are some issues there.

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy have you tried rebasing this with your latest changes yet?

**Comment by alexvaluyskiy** [comment]
Yep, no changes

**Comment by alexvaluyskiy** [comment]
@Aaronontheweb ready for review

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy got it!

File: src/core/Akka.Cluster.Tests.MultiNode/DeterministicOldestWhenJoiningSpec.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            }
+        }
+
+        [MultiNodeFact]
+        public void DeterministicOldestWhenJoiningSpecs()
+        {
+            Joining_cluster_must_result_in_deterministic_oldest_node();
+        }
+
+        public void Joining_cluster_must_result_in_deterministic_oldest_node()
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster.Tests.MultiNode/DeterministicOldestWhenJoiningSpec.cs]: Spec looks good

File: src/core/Akka.Cluster/ClusterDaemon.cs [hunk 2]
```diff
@@ -1162,8 +1162,15 @@ public void Joining(UniqueAddress node, ImmutableHashSet<string> roles)
...
                     _log.Info("Node [{0}] is JOINING, roles [{1}]", node.Address, string.Join(",", roles));
 
-                    // TODO: add deterministic behavior in https://github.com/akkadotnet/akka.net/pull/2101
-                    if (!node.Equals(SelfUniqueAddress))
+                    if (node.Equals(SelfUniqueAddress))
+                    {
+                        if (localMembers.IsEmpty)
+                        {
+                            // important for deterministic oldest when bootstrapping
+                            LeaderActions();
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/ClusterDaemon.cs]: Good

**Comment by Aaronontheweb** [comment]
Solid work. The changes make sense to me - I'll pull this into my work on the transition spec as well.

**Pull request merged**